cameraview/src/main/java/com/otaliastudios/cameraview/engine/Camera2Engine.java

```diff
@@ -1252,6 +1261,33 @@ public void setPlaySounds(boolean playSounds) {
         mPlaySoundsOp.end(null);
     }
 
+    @Override public void setPreviewFrameRate(float previewFrameRate) {
+        mPreviewFrameRate = previewFrameRate;
+        mHandler.run(new Runnable() {
+            @Override
+            public void run() {
+                if (getEngineState() == STATE_STARTED) {
+                    if (applyPreviewFrameRate(mRepeatingRequestBuilder)) {
+                        applyRepeatingRequestBuilder();
+                    }
+                }
+            }
+        });
+    }
+
+    @SuppressWarnings("WeakerAccess")
+    protected boolean applyPreviewFrameRate(@NonNull CaptureRequest.Builder builder) {
+        Collection<Range<Integer>> fpsRange = mCameraOptions.getSupportedFpsRange();
+        for (Range<Integer> range : fpsRange) {
+            if (range.getUpper() >= mPreviewFrameRate) {
+                mIsHigherFrameRateSupported = true;
+                builder.set(CaptureRequest.CONTROL_AE_TARGET_FPS_RANGE, range);
+                return true;
+            }
+        }
+        return false;
```

Comment:
We should address here the case of `mPreviewFrameRate == 0F` here before anything else. I think you can choose the first range in `characteristics.get(CONTROL_AE_AVAILABLE_TARGET_FPS_RANGES)`. Or choose the larger range... Basically a reasonable default. Then always return true.

If `!= 0F` your logic works well but I would replace `range.getUpper() >= mPreviewFrameRate` with `range.contains(mPreviewFrameRate)` so the lower bound is checked. Also you can remove mIsHigherFrameRateSupported as said above

Reply:
I think that if the previewFrameRate is 0 then we should use the Range from the min and max values calculated in CameraOptions. Because that would give it a default range with all the possibilities rather than finding the available ranges again and selecting one. We already have fetched the ranges in CameraOptions and calculated the min and max value.

Reply:
@vaibhavbhandula I think this can crash. If the camera returns [30, 60] and [60, 120], you will apply [30, 120]. But the range passed to `CONTROL_AE_TARGET_FPS_RANGE` should be one of those present in `CONTROL_AE_AVAILABLE_TARGET_FPS_RANGES`.

I think you should have separate logic. If `0F`, take the first range, or better, the first range that contains 30 or 24 (the most common values). This way we make sure that we do not default on something like [60, 120].

If not `0F`, I think taking the first one that `contains()` our value is OK.